### ISSUE TITLE
Research Formatted Guide Pages Parse with Markdown

### ISSUE THREAD

**Issue by abenipa3** [hackforla/website#1085]
### Overview
We need to research how to create formatted guide pages with Markdown so that the format could be reusable. 

### Action Items

Developer
- [x] Research if Jekyll can use Markdown files as a collection to populate unique files from a template.
- [x] Figure out how to use how to guide pages with sticky navigation. (Visit About Us page for reference.)
- [x] Create a working HTML file demo that pulls MD file.

Design
- [x] Once we have a feasibility plan for how to code it, consult with designers and product.

### Resources/Instructions
[Example of Markdown File of Role](https://github.com/hackforla/civic-opportunity/blob/master/roles-1/data-scientist.md)
Check the _layouts folder
https://jekyllrb.com/docs/step-by-step/04-layouts/
https://talk.jekyllrb.com/t/rendering-markdown-inside-an-html-include/4186/2

**Comment by akibrhast** [comment]
From @alyssabenipayo 
> Here's the Markdown Research for Guide Pages document. It's still a work in progress until the issue is closed. Please feel free to comment if needed. Thanks!

https://docs.google.com/document/d/1ivPbOLQh_KYmG7r6Fvx5Goojc3207gDxjTmDvT9vjQ8/edit

**Comment by abenipa3** [comment]
Progress: Documented findings from research in this [Google Doc](https://docs.google.com/document/d/1ivPbOLQh_KYmG7r6Fvx5Goojc3207gDxjTmDvT9vjQ8/edit) and added to the Hack for LA Drive. Ready for review.
Blocks: None yet.
Availability: Unavailable this week, but I'll be back next week (Week of 3/7). If anything, I am open to schedule a meeting.
ETA: 2/28/21

Drafted a Prototype below (Tested in private repo):
![image](https://user-images.githubusercontent.com/38295612/109429286-e4375000-79af-11eb-80af-48edfb949706.png)
Drafted Prototype Markdown Preview:
![image](https://user-images.githubusercontent.com/38295612/109429933-6f661500-79b3-11eb-8079-b71383770f5f.png)

**Comment by abenipa3** [comment]
2/28/21 - I approve of continuing to work on this issue. Added "Create a working HTML file demo that pulls MD file." to the description.

Availability: Unavailable this week, but I'll be back next week (Week of 3/7). If anything, I am open to schedule a meeting.

ETA: 3/13/21

**Comment by abenipa3** [comment]
Progress: N/A - As previously mentioned in the latest comment, I was unavailable last week.
Blocks: N/A - I will continue building the prototype this week.
Availability: Open 
ETA: 3/13/21

**Comment by abenipa3** [comment]
Progress: Built a working prototype as seen below. Will present during the meeting.
Blocks: Just need to revise the [document](https://docs.google.com/document/d/1ivPbOLQh_KYmG7r6Fvx5Goojc3207gDxjTmDvT9vjQ8/edit) of findings. 
Availability: Open
ETA: 3/14/21
![image](https://user-images.githubusercontent.com/38295612/111065879-46e11f00-8479-11eb-935f-c00a2f3fb697.png)

**Comment by abenipa3** [comment]
Live Preview of Prototype: https://github.com/alyssabenipayo/hello-world
Updated Document of Markdown Research: https://docs.google.com/document/d/1ivPbOLQh_KYmG7r6Fvx5Goojc3207gDxjTmDvT9vjQ8/edit#

**Comment by abenipa3** [comment]
@ExperimentsInHonesty and @daniellex0 -

Completed building the prototype. Quick Notes:
- Jekyll automatically converts MD files into HTML.
- We are able to create a navigation bar for guide pages. 

Please see [site preview here](https://alyssabenipayo.github.io/hello-world/) and the [repo for this prototype](https://github.com/alyssabenipayo/hello-world). 

I've also created a [document of the Markdown/Jekyll Research](https://docs.google.com/document/d/1ivPbOLQh_KYmG7r6Fvx5Goojc3207gDxjTmDvT9vjQ8/edit#heading=h.y67x7timd2s) and added it to the Hack for LA drive.

Here are quick previews of the prototypes:
![image](https://user-images.githubusercontent.com/38295612/111595873-d17f9200-8789-11eb-83a8-801310e2d313.png)
![image](https://user-images.githubusercontent.com/38295612/111595921-d8a6a000-8789-11eb-8c08-f9801da39fb8.png)
![image](https://user-images.githubusercontent.com/38295612/111595947-e1977180-8789-11eb-8d64-4a3b716affd8.png)

**Comment by daniellex0** [comment]
@alyssabenipayo Awesome!  🎉

**Comment by abenipa3** [comment]
Progress: [Completed building prototype](https://github.com/hackforla/website/issues/1085#issuecomment-801730535). Need to consult with Designers and Product.
Blocks: N/A
Availability: Open
ETA: 3/21/21

**Comment by abenipa3** [comment]
Feedback: Update the nav-bar so that it follows the users rather than freezes on the page. 

References:
[About Us](https://www.hackforla.org/about/#accomplishments
)
![image](https://user-images.githubusercontent.com/38295612/111915781-34488600-8a35-11eb-82bd-c1180d770973.png)

**Comment by abenipa3** [comment]
Progress: Able to create a sticky nav-bar. Need to clean up before it's ready.

<details><summary>PREVIEW</summary>

![image](https://user-images.githubusercontent.com/38295612/112746073-87549880-8f61-11eb-8f57-b555dba960ae.png)

</details>

Blocks: Got unexpectedly busy this past week so I was not able to finish the prototype as soon as expected.

Availability: Open

ETA: 3/30/21

**Comment by abenipa3** [comment]
@ExperimentsInHonesty @daniellex0 

I have updated the prototype with a sticky nav as seen below.

![Guide Page Prototype](https://user-images.githubusercontent.com/38295612/113092272-89f20080-91a2-11eb-8aa6-03862865a6f1.gif)


The  [Live Preview](https://alyssabenipayo.github.io/hello-world/github-issues) and [Repo](https://github.com/alyssabenipayo/hello-world) are available as well. Please feel free to let me know your thoughts.

**Comment by tarang100** [comment]
@alyssabenipayo let's discuss during Sunday's meeting. The nav seems to be a little off from the thing that it is supposed to be showing. Also font showing as Times.....all of this could be because of our browser choice. Lets review during the meeting

**Comment by akibrhast** [comment]
@tarang100 If I am not mistaken the spirit and the point of this issue was to research the possibility of  the feasibility on creating a page from markdown as well as a sticky navigation. 

I believe the video demonstrated by @alyssabenipayo clearly demonstrate that result of this research demonstrates that
- It is feasible to have formatted guide pages with markdown with sticky navigation


> The nav seems to be a little off from the thing that it is supposed to be showing. Also font showing as Times - @tarang100 

Since this is purely a demo, I don't think the font or the 'little of' is important. at this stage of the process

It would be best to close this issue now as we have concluded and resolved the research aspect of this issue while opening a new issue along the lines of:
`Implement Formatted Guide Pages With Markdown`

**Comment by abenipa3** [comment]
@tarang100, @akibrhast is correct. We were researching to see if it's feasible. However, I noticed the bug as well and meant to clean it up before Sunday's meeting.

I'm speaking with Danielle at the moment to see if she has any other feedback on this issue before we close it. Also clarifying to see if we should incorporate a task for the design team to design the page, too.

We've also created a new issue called [Create Formatted Guide Pages Parse with Markdown](https://github.com/hackforla/website/issues/1086). Thank you.

**Comment by abenipa3** [comment]
Progress: Updated prototype. See notes below. Need to consult with Designers and Product.
Blocks: N/A
Availability: Open
ETA: 4/6/21

<details> <summary>Notes Regarding Prototype:</summary> 

This Prototype is to show that it is feasible to have a formatted guide page with sticky navigation. 

WIPS:

- Need to find out how to add new sections into the navigation bar without needing to hard code. (t's possible to have Jekyll generate it automatically.)

- Navbar still contains bugs in the prototype- It jumps when scrolling and the arrow does not directly point to a section unless one of the other links has been selected.

</details>

**Comment by abenipa3** [comment]
Progress: Fixed bugs prototype. Will consult with Design and Product during tomorrow's meeting.
Blocks: N/A
Availability: Open
ETA: 4/11/21

- [Live Preview](https://alyssabenipayo.github.io/hello-world/github-issues) 
- [Repo](https://github.com/alyssabenipayo/hello-world)
- [Documentation of Research](https://docs.google.com/document/d/11y8YOuldNX-6ziKnDAmvIeDbKjGsdzqMad8DVu-8ewk/edit#heading=h.jgbt2e62ebpz)

![prototype-result](https://user-images.githubusercontent.com/38295612/114284776-1c966900-9a07-11eb-944f-e9c3ecb38e16.gif)

**Comment by abenipa3** [comment]
Feedback from 4/11/21's meeting:

- Consult with our Design Team to discuss:
   - Color for Navigation Bar Links/Design.
   - Mobile Design.

**Comment by abenipa3** [comment]
Progress: 

- Consulted with Design Team. 
- Added Subheadings and Table of Contents to the Prototype's Nav-Bar. 
- Currently designing a web version of the Guide Page. 

Blocks: N/A
Availability: Open
ETA: 4/24/21

- [Live Preview](https://alyssabenipayo.github.io/hello-world/github-issues) 
- [Repo](https://github.com/alyssabenipayo/hello-world)
- [Documentation of Research](https://docs.google.com/document/d/11y8YOuldNX-6ziKnDAmvIeDbKjGsdzqMad8DVu-8ewk/edit#heading=h.jgbt2e62ebpz)

**Comment by abenipa3** [comment]
**Progress:**
- Updated Prototype with Table of Contents to automate the process of nav-bar populating all headings without JavaScript. 
  - We also want to resolve the nav-bar arrow issue (Jumping/Skipping a section while scrolling).

<details><summary>Screenshot</summary> 

![image](https://user-images.githubusercontent.com/38295612/115989085-6b004780-a571-11eb-80d7-60a9b6522c94.png)

</details>

- Designed low-fidelity mockup of Guide Page. (Still a work in progress.)

<details><summary>Screenshot</summary> 

![image](https://user-images.githubusercontent.com/38295612/115989115-92efab00-a571-11eb-9f5a-ef83662e23c5.png)
</details>


**Blocks:** 

- Need to include an anchor so that the page scrolls to the correct place when the user hits the nav links.
  - Note: The goal is to avoid relying on JavaScript?
- Will also need to meet with Danielle regarding the Design System.

**Availability:** Open

**ETA:** 5/1/21

<details><summary> New References for Table of Contents Research</summary> 

[Jekyll Pure Liquid Table of Contents](https://github.com/allejo/jekyll-toc) - Thanks for the recommendation, Akib and Josh.
[Jekyll Table of Contents - Wikipedia Look!](https://blog.webjeda.com/jekyll-toc/)
[How to create a table of contents? -Jekyll ](https://stackoverflow.com/questions/9602936/how-to-create-a-table-of-contents-to-jekyll-blog-post)

</details>

- [Live Preview](https://alyssabenipayo.github.io/hello-world/github-issues) 
- [Repo](https://github.com/alyssabenipayo/hello-world)

**Comment by abenipa3** [comment]
Note: Will meet with Danielle on Tuesday at 3pm.

**Comment by abenipa3** [comment]
**Progress:**
- Discussed the Design System with Danielle + Gathered ideas for guide page designs.
- Able to include the arrow. Need to adjust it so it's inside the nav-bar box and that it highlights the sections when we scroll through the page.

<details><summary>Screenshot</summary> 

![image](https://user-images.githubusercontent.com/38295612/116810033-41599a00-aaf6-11eb-9238-e3bb14a1dd64.png)

</details>

**Blocks:** 

- Was feeling unwell this past week.

**Availability:** Open this week except 5/4 and 5/5.

**ETA:** 5/3/21

- [Live Preview](https://alyssabenipayo.github.io/hello-world/github-issues) 
- [Repo](https://github.com/alyssabenipayo/hello-world)

**Comment by abenipa3** [comment]
We have a feasible version of it. We are just waiting/working on the design to implement it on the site.
